index.html

```diff
@@ -262,29 +240,60 @@ <h2>The <dfn>WoT</dfn> object</h2>
 
     <section> <h3>The <dfn>discover()</dfn> method</h3>
       <p>
-        Starts the discovery process that will provide <code><a>ConsumedThing</a></code> objects that match the optional argument <code><a>ThingFilter</a></code>. When the argument is not provided, starts the widest discovery the <a>Thing Description</a> and <a>Protocol Bindings</a> allow.
-        Returns an [<code>Observable</code>](https://github.com/tc39/proposal-observable) object that can be subscribed and unsubscribed to.
+        Starts the discovery process that will provide <code><a>ConsumedThing</a></code> objects that match the optional argument <code><a>ThingFilter</a></code>. When the argument is not provided, starts the widest discovery the <a>Thing Description</a> and <a>Protocol Bindings</a> allow and support.
+        Returns an [<code>Observable</code>](https://github.com/tc39/proposal-observable) object that can be subscribed to and unsubscribed from.
       </p>
+
+      <section data-dfn-for="DiscoveryMethod">
+        <h4>The <dfn>DiscoveryMethod</dfn> enumeration</h4>
+        <pre class="idl">
+          typedef DOMString DiscoveryMethod;
+        </pre>
+        <p>
+          <a>DiscoveryMethod</a> represents the discovery type to be used:
+          <ul>
+            <li><dfn>"any"</dfn> does not provide any restriction</li>
+            <li><dfn>"local"</dfn> for discovering <a>Thing</a>s defined in the same device</li>
+            <li>
+              <dfn>"nearby"</dfn> for discovering <a>Thing</a>s nearby the device, e.g. by Bluetooth or NFC
+            </li>
+            <li>
+              <dfn>"directory"</dfn> for discovery based on a service provided by a directory or repository of <a>Thing</a>s
+            </li>
+            <li>
+              <dfn>"broadcast"</dfn> for an open ended discovery based on sending a request to a broadcast address
+            </li>
+            <li>
+              <dfn>"other"</dfn> for a proprietary method defined by the solution.
+            </li>
+          </ul>
+        </p>
+      </section>
+
       <section data-dfn-for="ThingFilter">
         <h4>The <dfn>ThingFilter</dfn> dictionary</h4>
         <p>
           The <a>ThingFilter</a> dictionary that represents the constraints for discovering <a>Thing</a>s as key-value pairs.
         </p>
         <pre class="idl">
           dictionary ThingFilter {
-            (DiscoveryMethod or DOMString) method = "any";
+            DiscoveryMethod method = "any";
             USVString url;
-            Dictionary description;
+            USVString query;
+            sequence&lt;Dictionary&gt; constraints;
```

Comment:
The AND and OR relationship for constraints seems to be very powerful but also complex at first. Moreover, the link to "the semantic metadata" in its description is very fuzzy. As I understand it, the key of the key-value pair is a reference to `SemanticType`. If so we might want to add an example.

Reply:
Example 4 is such an example: discover nearby things by Bluetooth OR NFC.